std/object/expressr.go

```diff
@@ -6,12 +6,13 @@ import (
 	"github.com/named-data/ndnd/std/utils"
 )
 
-// (advanced) express a single interest with reliability
+// Express a single interest with reliability
 func (c *Client) ExpressR(args ndn.ExpressRArgs) {
-	c.outpipe <- args
+	ExpressR(c.engine, args)
 }
 
-func (c *Client) expressRImpl(args ndn.ExpressRArgs) {
+// Express a single interest with reliability
+func ExpressR(engine ndn.Engine, args ndn.ExpressRArgs) {
 	sendErr := func(err error) {
 		args.Callback(ndn.ExpressCallbackArgs{
```

Comment:
If `args.Callback` is called directly in `engine.Express`'s callback, it must be non-blocking. I think we need to mention this in the comment:
https://github.com/named-data/ndnd/blob/0c6a9a61475284aaedff5e7cfc9c27f725e46e47/std/ndn/client.go#L98-L99

Reply:
Clarified.